x-pack/plugins/siem/server/routes/index.ts

```diff
@@ -39,27 +39,28 @@ export const initRoutes = (
   router: IRouter,
   config: ConfigType,
   usingEphemeralEncryptionKey: boolean,
-  security: SetupPlugins['security']
+  security: SetupPlugins['security'],
+  ml: SetupPlugins['ml']
 ) => {
   // Detection Engine Rule routes that have the REST endpoints of /api/detection_engine/rules
   // All REST rule creation, deletion, updating, etc......
-  createRulesRoute(router);
+  createRulesRoute(router, ml);
   readRulesRoute(router);
-  updateRulesRoute(router);
-  patchRulesRoute(router);
+  updateRulesRoute(router, ml);
+  patchRulesRoute(router, ml);
   deleteRulesRoute(router);
   findRulesRoute(router);
 
   addPrepackedRulesRoute(router);
```

Comment:
Prepackaged rules will contain ML types, do you want to add ML checks and block adding _any_ rule additions from occurring if the user does not have privileges or maybe we have to do something more involved with it where only the non-ML rules are processed?

I am fine if you don't add it for now, but maybe just a TODO or NOTE above this as a reminder for down the road?

Reply:
I believe this one was unguarded previously because the prepackaged rule still has to be enabled after installation.


This PR is meant to prevent non-ML admins from creating/editing enabled ML rules. I could go either way on this one, but my inclination is just to guard against situations where a non-ML admin could _break_ the execution of an ML Rule. If in the future alerting gives us a way to "persist" the rule creators permissions, most of these guards will become unnecessary (and in their current form would even break the use case [craig described above](https://github.com/elastic/kibana/pull/65583#issuecomment-624973777)).